model/session.go

```diff
@@ -56,11 +57,13 @@ func GetDeviceSubject(tenantID, deviceID string) string {
 
 // Session represents a session from a user to a device and its attributes
 type Session struct {
-	ID       string    `json:"id" bson:"_id"`
-	UserID   string    `json:"user_id" bson:"user_id"`
-	DeviceID string    `json:"device_id" bson:"device_id"`
-	StartTS  time.Time `json:"start_ts" bson:"start_ts"`
-	TenantID string    `json:"tenant_id" bson:"-"`
+	ID                 string      `json:"id" bson:"_id"`
+	UserID             string      `json:"user_id" bson:"user_id"`
+	DeviceID           string      `json:"device_id" bson:"device_id"`
+	StartTS            time.Time   `json:"start_ts" bson:"start_ts"`
+	TenantID           string      `json:"tenant_id" bson:"-"`
+	BytesRecordedMutex *sync.Mutex `json:"-" bson:"-"`
```

Comment:
Potential NULL pointer exception
```suggestion
	BytesRecordedMutex sync.Mutex `json:"-" bson:"-"`
```

Reply:
basically no-can-do, you cant pass the mutex by value, and it is passed to the validator, among others, and it makes static checker go crazy. I kind of always use pointers to mutexes, do you have a strong opinion here? Where would the NPR happen? (apart from the fact that we need a central place to create a new session, I checked all of them and added the mutex)

Reply:
OK, no strong opinion.